File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NoServerErrorDialog.java
package it.dani.selfhomeapp.middle.view.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.function.Supplier;

import it.dani.selfhomeapp.R;

public class NoServerErrorDialog extends DialogFragment {

    private Supplier<Void> positiveSupplier;
    private Supplier<Void> negativeSupplier;

    public NoServerErrorDialog()
    {
        super();
        this.positiveSupplier = null;
        this.negativeSupplier = null;
    }

    public NoServerErrorDialog(@NonNull Supplier<Void> positiveSupplier, @NonNull Supplier<Void> negativeSupplier)
    {
        super();
        this.positiveSupplier = positiveSupplier;
        this.negativeSupplier = negativeSupplier;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstance)
    {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setTitle(R.string.titleHandleNoServerResponseErr);
        builder.setMessage(R.string.noServerResponseErr);
        builder.setPositiveButton(R.string.positiveHandleNoServerResponseErr,(x,y) -> {
            if(this.positiveSupplier != null) this.positiveSupplier.get();
        });
        builder.setNegativeButton(R.string.negativeHandleNoServerResponseErr,(x,y) -> {
            if(this.negativeSupplier != null) this.negativeSupplier.get();
        });

        return builder.create();
    }
}

File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NoServerErrorDialog.kt
package it.dani.selfhomeapp.middle.view.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import it.dani.selfhomeapp.R

class NoServerErrorDialog(private val onPositive : () -> Any, private val onNegative : () -> Any, private val onNeutral: () -> Any) : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(this.activity).apply {
            setTitle(R.string.titleHandleNoServerResponseErr)
            setMessage(R.string.noServerResponseErr)
            setPositiveButton(R.string.positiveHandleNoServerResponseErr) {_,_ ->
                this@NoServerErrorDialog.onPositive()
            }
            setNegativeButton(R.string.negativeHandleNoServerResponseErr) {_,_ ->
                this@NoServerErrorDialog.onNegative()
            }
            setNeutralButton(R.string.neutralHandleNoServerResponseErr) {_,_ ->
                this@NoServerErrorDialog.onNeutral()
            }
        }.create()
    }
}